src/components/Toast/DefaultToast/index.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+
+import styles from '@/components/Toast/DefaultToast/DefaultToast.css';
+import IconClose from '@/assets/images/icons/web/close.svg';
+
+interface Props {
+  className?: string;
+  autoClose?: number;
+  title?: string;
+  message?: string;
+  onButtonClick: () => void;
+  size?: string;
+  variablesClassName?: string;
+  getIcon?: () => React.ReactNode;
+  getCloseButton?: () => React.ReactNode;
+  getBody?: () => React.ReactNode;
+}
+
+const DefaultToast: React.FC<Props> = props => {
+  const {
+    className,
+    autoClose,
+    title,
+    message,
+    onButtonClick,
+    size,
+    getIcon,
+    getCloseButton,
+    getBody,
+    variablesClassName,
+    ...toastProps
+  } = props;
+
+  const sizeClass = `toast--${size}`;
+  const sizeMessageClass = `toast-body-message--${size}`;
+  const sizeTitleClass = `toast-body-title--${size}`;
+  const sizeIcon = 10;
+  const [hideMessage, setHideMessage] = useState(false);
+
+  const handleCloseClick = () => {
+    setHideMessage(true);
+  };
+
+  useEffect(() => {
+    if (autoClose) {
+      setTimeout(() => {
+        setHideMessage(true);
+      }, autoClose);
+    }
+  }, []);
+
+  return (
+    <div
+      {...toastProps}
+      className={classNames(
+        className,
+        sizeClass,
+        variablesClassName,
+        styles.toast,
+        styles[sizeClass],
+        hideMessage ? styles['toast-hidden'] : ''
+      )}
+    >
+      {getIcon ? getIcon() : ''}
+      {getBody ? (
+        getBody()
+      ) : (
+        <div className={styles['toast-body']}>
+          {title && (
+            <p className={classNames(styles['toast-body-title'], styles[sizeTitleClass])}>
+              {title}
+            </p>
+          )}
+          {message && (
+            <span className={classNames(styles['toast-body-message'], styles[sizeMessageClass])}>
+              {message}
+            </span>
+          )}
+        </div>
+      )}
+      {getCloseButton ? (
+        getCloseButton()
+      ) : (
+        <button
+          onClick={onButtonClick || handleCloseClick}
```

Comment:
is there a reason to call one or the other? the user is clicking in close we would for sure need to close it nonetheless, but we could also call the close function the user engineer is passing
something like 

```
  const handleCloseClick = () => {
    setHideMessage(true);
    onButtonCLick()
  };
```

Reply:
it makes sense, thanks